src/backend/web/graphql/index.js

```diff
@@ -113,37 +113,56 @@ module.exports.resolvers = {
      * @return Array of 'perPage' number of Post objects
      */
     getPosts: async (parent, { filter, page, perPage }) => {
+      let postIds;
+      const numOfPosts = await getPostsCount();
       const prPage = perPage > maxPostsPerPage ? maxPostsPerPage : perPage;
 
-      const numOFPosts = await getPostsCount();
-      const first = page * prPage;
-
-      if (first < numOFPosts) {
-        const last = first + prPage < numOFPosts ? first + prPage : numOFPosts;
-        const postIds = await getPosts(first, last);
-
+      // For Date Filter
+      if (filter) {
```

Comment:
Flip this logic: `if (!filter) {...}` since the code for the other case is smaller, and then you can `return` and not indent all this.

Even better, split this out into a separate function.

Reply:
Opting to do this in another PR.